pkg/gcp/ocpgwdeployer.go

```diff
@@ -251,7 +251,9 @@ func (d *ocpGatewayDeployer) deployGateway(zone string) error {
 	}
 
 	if d.image == "" {
-		d.image, err = d.msDeployer.GetWorkerNodeImage(machineSet, d.InfraID)
+		workerNodeList := []string{d.InfraID + "-worker-b", d.InfraID + "-worker-c", d.InfraID + "-worker-d"}
```

Comment:
Can we safely assume that nodes will be labelled in this manner (i.e., `-worker-x`)?
Also, here we are hard-coding the list to just four nodes. Is there a better way to simply query the worker nodes based on the labels and pick one of them?

Reply:
I'm not sure I follow. Why what was done? Lines 254-255 are new, added by this PR. Can you be more specific re: a `ListAll` method in admiral, ie specifically where would it be added and what would it do?

Reply:
These are moved from the pkg/ocp/machinesets.go to this file and now the workerNodeList is passed as argument  to msDeployer.GetWorkerNodeImage so that the method can be reused. This may be referring to a list all function to see all the worker nodes in the infra.

Reply:
OK I still don't see where such a `ListAll` function would go or why it would be needed in admiral. Not trying to nit but unless we understand what the TODO means, I'd rather not propagate it.

Reply:
I removed the TODO.

Reply:
Basically, we are using the admiral dynamic Interface API to Get the machineSet details associated with the worker-node. Currently, the admiral API does not have the listAll method (https://github.com/submariner-io/admiral/blob/69a40598bdd63f570d0e695b2fd1a3ebdeb8f0f0/pkg/resource/interface.go#L28), hence we are using this approach to get the machineSet details of the associated worker nodes. If we have a listAll method in admiral, we can remove this hard-coding of the workerNodeList.

@aswinsuryan thanks for updating the PR. It looks good. Can you please add a TODO with the above notes so that once Admiral is enhanced we can update this code accordingly?

Reply:
I'm confused. I don't see the admiral `Resource` interface being used anywhere in this code path. This code came from `k8sMachineSetDeployer.GetWorkerNodeImage` which uses a `dynamic.Interface` obtained from

    `machineSetClient, err := msd.clientFor(machineSet)`

 so the standard K8s List method is available.

Reply:
Yes @tpantelis. I am not able to recollect the exact reason why that TODO was added. But it looks like we can use machineSetClient.List() method to get the list of worker nodes and remove the hardcoded list.

Reply:
I have added a TODO and raised this issue to track https://github.com/submariner-io/cloud-prepare/issues/201